### DISCUSSION TITLE
Any way to include dislike count on Invidious instances?

### DISCUSSION THREAD

**Discussion by Nikoichu** [Anarios/return-youtube-dislike#529] [General]
I'm trying to transition over to invidious and it would be useful to have my dislike counter to spot fake tutorials and such. Is there currently a way to do it, and is it a planned feature?

**Comment by sy-b** [comment]
yes

take a look here https://github.com/Anarios/return-youtube-dislike#api-documentation

**Reply by Nikoichu** [reply]
Hey thanks for the link, but could you please be more specific? I don't really understand a lot about how to use APIs, since I'm no programmer, I'm just a regular user that's trying to minimize Youtube's bullshittery 😄

**Reply by sy-b** [reply]
There are 2 options

1. Invidious maintainer(s) will have to add it themselves
2. Make another extension for invidious

Technically, we can add support for invidious but that'll require us to add another "site permission". Basically, we have to declare to the browser which site(s) this extension is going to access. Now, when the extension updates in the background & if the browser finds that the newer version requires more permissions, it disables the extension & asks users for permission before continuing. 

![unknown](https://user-images.githubusercontent.com/94835959/163735447-6234cda1-f483-482b-976c-3095f9de83ed.png)

This popup kind of encourages users to click "Remove" which obviously uninstalls the extension.

In January, a feature was added to test if the extension has stopped working (because of a bug in chromium browsers). This required site permission to returnyoutubedislike.com .  This change was quickly reversed (within a few hours). But the damage was done. The extension lost about 100k users. This number was significant back then (it still is).

So, I don't think we'll add support for any other site in this extension (**probably** unless chrome add the "optional_host_permissions" feature)

**Reply by Nikoichu** [reply]
I see. I don't know about chrome or chromium based browsers, but with sponsorblock for firefox, you can manually add more domains in the addon settings menu, and it will ask for the bonus permissions then. I don't think that would be scary for any users, because they added the option themselves. You could take a look at how they did it: https://github.com/ajayyy/SponsorBlock

**Reply by sy-b** [reply]
SponsorBlock has permissions for 3 sites
See chrome://extensions/ --> [extension] --> Details --> Permissions --> Site access

That domain thing is for other purpose. The permissions I am referring is to **read site pages**. That is - when users visits that page, the extension will be able to read & edit the page. For e.g. how RYD & SB do for YT.

It is possible to get access to all sites but that again will require permission modification & that eventually that pop up.

**Reply by Nikoichu** [reply]
Yes, I'm talking about those same permissions. In Firefox they look like this:
"X extension wants to access your data for the following sites: SomeStupidDomain.com".

Again, I don't know if Chrome has that functionality, but Firefox only asks for those bonus permissions once you enable that setting:
![firefox_Bl3t7zvZMm](https://user-images.githubusercontent.com/42444745/163788116-0c7dd47e-88e1-4867-85bc-e3bbb2ec301c.png)

After you click "Add", comes the scary popup, that shouldn't be scary to people who know what they're doing:

![ZDaYFH1bRb](https://user-images.githubusercontent.com/42444745/163788126-273a311d-cc82-449b-a4ac-842b3a22a013.png)

**Reply by sy-b** [reply]
I was not able to find that in chrome.

But here in it's `manifest.json` (used to declare info about the extenion such as name, description, version, permissions etc) I found this interesting thing.
![chrome manifest declarativeContest](https://user-images.githubusercontent.com/94835959/163916207-ed73b89d-daf4-4df0-94c6-9cc55749945d.PNG)
- SponsorBlock is in chrome manifest v2. RYD is in v3. The problem with v3 is that it doesn't accept URLs in `optional_permissions`. And chrome webstore doesn't allow to downgrade.
- `*://*/*` means that SB can access all sites. But line is not present in FireFox version
- I wasn't aware of `declarativeContent`. Probably this might help.
 
 Thanks 😀

**Reply by Nikoichu** [reply]
Glad I was able to kinda help 😅
Let me know if I can do anything to help further (test something or whatnot). ❤️

**Comment by Chirunos** [comment]
I'm using [this](https://github.com/jesperbakhandskemager/Return-Invidious-Dislike-Count) user script as a workaround
